Content.Shared/RepulseAttract/RepulseAttractSystem.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Content.Shared.Physics;
+using Content.Shared.Throwing;
+using Content.Shared.Timing;
+using Content.Shared.Weapons.Melee.Events;
+using Content.Shared.Whitelist;
+using Content.Shared.Wieldable;
+using Robust.Shared.Map;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Components;
+using System.Numerics;
+
+namespace Content.Shared.RepulseAttract;
+
+public sealed class RepulseAttractSystem : EntitySystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly ThrowingSystem _throw = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
+    [Dependency] private readonly SharedTransformSystem _xForm = default!;
+    [Dependency] private readonly UseDelaySystem _delay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RepulseAttractComponent, AttemptMeleeEvent>(OnMeleeAttempt, after: [typeof(SharedWieldableSystem)]);
+    }
+    private void OnMeleeAttempt(Entity<RepulseAttractComponent> ent, ref AttemptMeleeEvent args)
+    {
+        if (args.Cancelled)
+            return;
+
+        if (_delay.IsDelayed(ent.Owner))
+            return;
+
+        TryRepulseAttract(ent, args.User);
+    }
+
+    public bool TryRepulseAttract(Entity<RepulseAttractComponent> ent, EntityUid user)
+    {
+        var position = _xForm.GetMapCoordinates(ent.Owner);
+        return TryRepulseAttract(ent.Comp.Attract, position, ent.Comp.Speed, ent.Comp.Range, ent.Comp.Whitelist, ent.Comp.Blacklist);
+    }
+
+    public bool TryRepulseAttract(bool attract, MapCoordinates position, float speed, float range, EntityWhitelist? whitelist = null, EntityWhitelist? blacklist = null)
+    {
+        var entsInRange = _lookup.GetEntitiesInRange(position, range, flags: LookupFlags.Dynamic | LookupFlags.Sundries);
```

Comment:
Same here, you should be re-using a set instead of getting one every time. Also adjust the flags to make the static check not required.

Reply:
To get this to work on melee attempt then we'd have to call getting ents in range on the melee check instead. 
We could make it a parameter instead but then we'd need to think of where we'd move the ents in range call.